### PR TITLE
Chore: import of app community-backend into monorepo

### PR DESCRIPTION
**Pull request by rjonczy**
- migrated community-backend
- added ci for community-backend
- dummy commit to trigger ci
- test / debug circle ci run
- trigger
- change node version

### CODE REVIEW

File: .circleci/workflows.yml [hunk 1]
```diff
@@ -469,6 +503,22 @@ workflows:
                 - develop
           context:
             - K8S
+  community-backend:
+    when:
+      or:
+        - << pipeline.parameters.run-community-backend-service-workflow >>
+        - << pipeline.parameters.ui-package-modified >>
```
Comment (rattrap) [inline .circleci/workflows.yml:510]: We don't depend on the UI for this

File: services/community-backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
+FROM node:14
+WORKDIR /app
+
+ENV PATH /app/node_modules/.bin:$PATH
+
+COPY package.json yarn.lock ./
+COPY services/community-backend/package.json /app/services/community-backend/
+
+COPY package.json yarn.lock /app/
```
Comment (rattrap) [inline services/community-backend/Dockerfile:9]: Why do we do this twice?

File: services/community-backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+COPY services/claim-backend/package.json /app/services/claim-backend/
```
Comment (rattrap) [inline services/community-backend/Dockerfile:10]: Don't need the claim here

File: services/community-backend/extensions/users-permissions/controllers/Auth.js [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+const formatError = error => [
+  { messages: [{ id: error.id, message: error.message, field: error.field }] },
+];
+
+
+
+module.exports = {
+  async forgotPassword(ctx) {
+    let { email, token } = ctx.request.body;
+    const gres = await axios.post(`https://www.google.com/recaptcha/api/siteverify?secret=6LdLJXAaAAAAALsfpslEdIBHpW-OYKPuUwFT2lbY&response=${token}`)
```
Comment (rattrap) [inline services/community-backend/extensions/users-permissions/controllers/Auth.js:15]: We should move this secret to an ENV var

File: services/community-backend/extensions/users-permissions/controllers/Auth.js [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+          );
+        }
+    
+        const params = {
+          ..._.omit(ctx.request.body, ['kyc', 'points', 'confirmed', 'confirmationToken', 'resetPasswordToken']),
+          provider: 'local',
+        };
+
+        if(!params.skipRecaptcha) {
+          const gres = await axios.post(`https://www.google.com/recaptcha/api/siteverify?secret=6LdLJXAaAAAAALsfpslEdIBHpW-OYKPuUwFT2lbY&response=${params.token}`)
```
Comment (rattrap) [inline services/community-backend/extensions/users-permissions/controllers/Auth.js:148]: Here too. Move secret to env var

**Review by rattrap — CHANGES_REQUESTED** [review]

File: .circleci/workflows.yml (hunk 1 above)
Comment (rjonczy) [reply .circleci/workflows.yml]: removed

File: services/community-backend/Dockerfile (hunk 3 above)
Comment (rjonczy) [reply services/community-backend/Dockerfile]: fix. btw in similar situation (duplication) is in claim-backend: https://github.com/Taraxa-project/taraxa-platform/blob/develop/services/claim-backend/Dockerfile#L9-L13

File: services/community-backend/Dockerfile (hunk 2 above)
Comment (rjonczy) [reply services/community-backend/Dockerfile:6]: fix. btw similar situation (duplication) is in claim-backend: https://github.com/Taraxa-project/taraxa-platform/blob/develop/services/claim-backend/Dockerfile#L9-L13

File: services/community-backend/extensions/users-permissions/controllers/Auth.js (hunk 5 above)
Comment (rjonczy) [reply services/community-backend/extensions/users-permissions/controllers/Auth.js]: @VargaElod23 handled pulling config from ENV and I have added `Secret` to helm-chart.

File: services/community-backend/extensions/users-permissions/controllers/Auth.js (hunk 4 above)
Comment (rjonczy) [reply services/community-backend/extensions/users-permissions/controllers/Auth.js]: @VargaElod23 handled pulling config from ENV and I have added `Secret` to helm-chart.

File: .circleci/workflows.yml [hunk 6]
```diff
@@ -78,6 +84,9 @@ aliases:
         if [ "${CIRCLE_BRANCH}" == "develop" ]; then
           export ENV=stage
           echo 'export ENV=stage' >> $BASH_ENV
+        elif [ "${CIRCLE_BRANCH}" == "chore/import-community-backend" ]; then
```
Comment (rattrap) [inline .circleci/workflows.yml:87]: Don't forget to remove this before merging.

File: .circleci/workflows.yml [hunk 7]
```diff
@@ -469,6 +509,41 @@ workflows:
...
+      or:
+        - << pipeline.parameters.run-community-backend-service-workflow >>
+    jobs:
+      - build-community-backend:
+          filters:
+            branches:
+              only:
+                - main
+                - develop
+                - chore/import-community-backend
```
Comment (rattrap) [inline .circleci/workflows.yml:523]: And this.

File: services/community-backend/Dockerfile (hunk 2 above)
Comment (rattrap) [reply services/community-backend/Dockerfile:6]: Hm. Yeah. Noted. We should fix that too

**Review by rattrap — APPROVED** [review]
LGTM

**Pull request merged**